web/packages/design/src/SlideTabs/SlideTabs.tsx

```diff
@@ -190,6 +210,18 @@ type FullTabSpec = TabContentSpec & {
    * attribute set to "tabpanel".
    */
   controls?: string;
+  tooltipContent?: React.ReactNode;
+  tooltipPosition?: Position;
+  /**
+   * An icon that will be displayed on the side. The layout will stay the same
+   * whether the icon is there or not. If `isProcessing` prop is set to `true`,
+   * the icon for an active tab is replaced by a spinner.
+   */
+  statusIcon?: React.ComponentType<IconProps>;
+  /** Color of the status icon. */
+  statusIconColor?: string;
+  /** Color of the status icon on an active tab. */
+  statusIconColorActive?: string;
```

Comment:
Also, do you think we need the `statusIcon` to be that configurable? 
I believe we could replace it with `status?: 'warning' | 'error'` and let the component figure out what icon to use and how to style it. 
It seems that we place too much responsibility on the callsites to provide correct values. Changing the colors of active or inactive tabs would require taking into account `statusIconColor` and `statusIconColorActive` which this component doesn't control. Instead, all usages would need to be inspected manually.

Reply:
> Also, do you think we need the `statusIcon` to be that configurable?
> I believe we could replace it with `status?: 'warning' | 'error'` and let the component figure out what icon to use and how to style it.
> It seems that we place too much responsibility on the callsites to provide correct values. Changing the colors of active or inactive tabs would require taking into account `statusIconColor` and `statusIconColorActive` which this component doesn't control. Instead, all usages would need to be inspected manually.

I was actually going to do it this way, but then I realized that the design I'm implementing calls out for a quite specific functionality of only showing status icons on a non-active tab, which basically boils down to giving it a `statusIconColorActive` set to `'transparent'`. Now, I could, obviously, reflect it here, and make it the only choice, but then I felt like I would be simply shifting a specific usage too much into the generic component. So yes, I can just implement it as you suggest, but with a caveat that these icons are not going to be shown on the active tabs.

Reply:
>I was actually going to do it this way, but then I realized that the design I'm implementing calls out for a quite specific functionality of only showing status icons on a non-active tab, which basically boils down to giving it a statusIconColorActive set to 'transparent'.

But since you have information about the active tab in the parent component, you can dynamically decide whether to display the status, based on the active tab. What do you think?
I mean something like this:
```ts
{
   key: 'tab-1',
   title: 'Tab',
   status: activeIndex === 0 ? undefined : tabStatus,
},
```

Reply:
@gzdunek That will work on a fundamental level, but the devil is in the details. Using transparent as icon color gives us an ability to transition between visible and invisible icon. If we don't do it, but we animate the slider and the text color, the effect is a bit awkward. So in order to do it, we'll need to remember the current status as we progress from a colored icon to the null state, in which case, the `status` field will need to be accompanied by a `statusVisible` flag (or, as it would require to be `true` by default, perhaps a `statusHidden` flag that defaults to `false`).

Reply:
(Which I'm fine with, by the way, I'm just stating that it needs to be done.)

Reply:
And even then, since it's animated, it semantically ties the status to the fact that we switch tabs, which means it would be animated _always_. Which, ultimately, suggests that we should just add another property on the top level, something like `hideStatusIconOnActiveTab` perhaps.

Reply:
>Which, ultimately, suggests that we should just add another property on the top level, something like hideStatusIconOnActiveTab perhaps.

This sounds reasonable to me.